src/lasdi/param.py

```diff
@@ -40,12 +47,18 @@ def __init__(self, config):
         for param in self.param_list:
             self.param_name += [param['name']]
 
-        self.train_space = self.createInitialTrainSpace(self.param_list)
-        self.n_init = self.train_space.shape[0]
-
         test_space_type = parser.getInput(['test_space', 'type'], datatype=str)
         if (test_space_type == 'grid'):
+            self.train_space = self.createInitialTrainSpace(self.param_list)
+            self.n_init = self.train_space.shape[0]
+
             self.test_grid_sizes, self.test_meshgrid, self.test_space = self.createTestGridSpace(self.param_list)
+        if (test_space_type == 'hull'):
+            assert self.n_param >=2, 'Must have at least 2 parameters if test_space is \'hull\' '
```

Comment:
Yes! I was previously using matplotlib's `path` to check if points were in the hull, which only works in 2D. However, we now mesh the training space with `Delaunay' so it works in arbitrary dimensions (although it will be very slow in higher dimensions). For example, I attached a picture the test/train space if we define the hull to be cut out from a cube with

parameter_space:
  parameters:
    - name: x
      min: 0.
      max: 1.
      sample_size: 21
      list: [0, 0, 0, 0, 1, 1, 1, 1, 0.5, 0.5]
      log_scale: false
    - name: y
      min: 0.
      max: 1.
      sample_size: 21
      list: [0, 1, 0, 1, 0, 1, 0, 1, 0, 1]
      log_scale: false
    - name: z
      min: 0.
      max: 1.
      sample_size: 21
      list: [0, 0, 0.1, 0.2, 0, 0, 1, 1, 0.5, 0.5]
      log_scale: false
  test_space:
    type: hull

![ramp](https://github.com/user-attachments/assets/825f4e83-c87d-4242-a9ab-dfad71895530)